api/services/gameService.js

```diff
@@ -34,43 +34,33 @@ async function fetchSpectatorUsernames(gameId) {
 
 // Used to create fully populated game
 function tempUser(usr, points) {
-  this.pNum = usr.pNum;
-  this.hand = usr.hand;
-  this.points = points;
-  this.faceCards = usr.faceCards;
-  this.frozenId = usr.frozenId;
-  this.id = usr.id;
-  this.username = usr.username;
-
-  this.hand.sort(comapreByRankThenSuit);
-  this.points.sort(comapreByRankThenSuit);
-  this.faceCards.sort(comapreByRankThenSuit);
+  Object.entries(usr).forEach(([key,val]) => {
+    if(key === 'encryptedPassword'){
+      return;
+    }
+    if(key === 'points'){
+      this[key] = points;
+      return;
+    }
+    this[key] = val;
+  });
+  this.hand.sort(compareByRankThenSuit);
+  this.points.sort(compareByRankThenSuit);
+  this.faceCards.sort(compareByRankThenSuit);
```

Comment:
The issue states 'note that this can't be done for tempPlayer as the player object needs to have at least the encryptedPassword field omitted'. I believe by just returning on the encrypted password it will omit it while still programmatically grabbing the rest.